client/sections.js

```diff
@@ -22,21 +22,20 @@ const extensionSections = extensions.map( extension => {
 	}
 } );
 
-if ( config.isEnabled( 'devdocs' ) ) {
-	sections.push( {
-		name: 'devdocs',
-		paths: [ '/devdocs' ],
-		module: 'devdocs',
-		secondary: true,
-		enableLoggedOut: true
-	} );
+sections.push( {
```

Comment:
How does it differ from using feature flags nested deeper into a chunk? I think it is a very good simplification that bring lots of benefits simplifying build process and drastically reducing its time.

However we need to sync it with extensions. I double checked it and it turns out that our code depends on `env_id` (https://github.com/Automattic/wp-calypso/blob/master/client/sections.js#L17). We plan to use `production` env for all non-development envs when preparing build targeted for Docker, which means it won't work as expected for `staging`, `wpcalypso` and `horizon`.